// @nx/expo/src/utils/versions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.babelRuntimeVersion = exports.jestExpoVersion = exports.testingLibraryJestNativeVersion = exports.testingLibraryReactNativeVersion = exports.reactNativeSvgVersion = exports.reactNativeSvgTransformerVersion = exports.expoMetroRuntimeVersion = exports.expoMetroConfigVersion = exports.reactNativeWebVersion = exports.reactNativeVersion = exports.typesReactVersion = exports.reactTestRendererVersion = exports.reactDomVersion = exports.reactVersion = exports.babelPresetExpoVersion = exports.expoCliVersion = exports.expoStatusBarVersion = exports.expoSplashScreenVersion = exports.expoVersion = exports.nxVersion = void 0;
exports.nxVersion = require('../../package.json').version;
exports.expoVersion = '~52.0.11';
exports.expoSplashScreenVersion = '~0.29.13';
exports.expoStatusBarVersion = '~2.0.0';
exports.expoCliVersion = '~0.21.8'; // @expo/cli
exports.babelPresetExpoVersion = '~12.0.1';
exports.reactVersion = '~18.3.1';
exports.reactDomVersion = '~18.3.1';
exports.reactTestRendererVersion = '~18.3.1';
exports.typesReactVersion = '~18.3.12';
exports.reactNativeVersion = '0.76.3';
exports.reactNativeWebVersion = '~0.19.13';
exports.expoMetroConfigVersion = '~0.19.4';
exports.expoMetroRuntimeVersion = '~4.0.0';
exports.reactNativeSvgTransformerVersion = '~1.5.0';
exports.reactNativeSvgVersion = '~15.8.0';
exports.testingLibraryReactNativeVersion = '~12.9.0';
exports.testingLibraryJestNativeVersion = '~5.4.3';
exports.jestExpoVersion = '~52.0.2';
exports.babelRuntimeVersion = '~7.26.0';
